go/arrow/flight/flightsql/client.go

```diff
@@ -243,6 +243,87 @@ func (c *Client) ExecuteSubstraitUpdate(ctx context.Context, plan SubstraitPlan,
 	return updateResult.GetRecordCount(), nil
 }
 
+// ExecuteIngest is for executing a bulk ingestion and only returns the number of affected rows.
+// The provided RecordReader will be retained for the duration of the call, but it is the caller's
+// responsibility to release the original reference.
+func (c *Client) ExecuteIngest(ctx context.Context, rdr array.RecordReader, reqOptions *ExecuteIngestOpts, opts ...grpc.CallOption) (int64, error) {
+	var (
+		err          error
+		desc         *flight.FlightDescriptor
+		stream       pb.FlightService_DoPutClient
+		wr           *flight.Writer
+		res          *pb.PutResult
+		updateResult pb.DoPutUpdateResult
+	)
+
+	// Servers cannot infer defaults for these parameters, so we validate the request to ensure they are set.
+	if reqOptions.TableDefinitionOptions == nil {
+		return 0, fmt.Errorf("cannot ExecuteIngest: invalid ExecuteIngestOpts, TableDefinitionOptions is required")
+	}
+	if reqOptions.Table == "" {
+		return 0, fmt.Errorf("cannot ExecuteIngest: invalid ExecuteIngestOpts, Table is required")
+	}
+
+	rdr.Retain()
+	defer rdr.Release()
```

Comment:
since you aren't passing the reader to anything outside of this function, I don't think this is necessary at all